src/lib/forms/components/LocationDatePicker.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { DatePicker } from '@components/DatePicker';
+import { locationApi } from '@api/locations';
+import { fromISO, toISODate } from '@api/date';
+import _isEmpty from 'lodash/isEmpty';
+
+export class LocationDatePicker extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      isLoading: false,
+      error: {},
+    };
+  }
+
+  fetchLocation = async locationPid => {
+    this.setState({ isLoading: true });
+    try {
+      const response = await locationApi.get(locationPid);
+      this.setState({ data: response.data, isLoading: false, error: {} });
+    } catch (error) {
+      this.setState({ isLoading: false, error: error });
+    }
+  };
+
+  listDisabled = () => {
+    const { minDate, maxDate } = this.props;
```

Comment:
Just IMHO, can we refactor/split in functions to be more readable and understandable?
I kind of feel scared to the idea of reading this wall of code! :D :)